axelor-quality/src/main/java/com/axelor/apps/quality/service/QIAnalysisServiceImpl.java

```diff
@@ -35,4 +97,168 @@ public int setAdvancement(QIAnalysis qiAnalysis) {
         qiTasksList.stream().map(QITask::getAdvancement).reduce((x, y) -> (x + y)).orElse(0);
     return totalAdvancement / qiTasksList.size();
   }
+
+  @Override
+  public List<QIActionDistribution> generateQIActionDistribution(QIAnalysis qiAnalysis)
+      throws AxelorException, IOException {
+
+    List<QITask> qiTasksList =
+        qiAnalysis.getQiTasksList().stream()
+            .filter(line -> line.isSelected())
+            .collect(Collectors.toList());
+
+    QualityConfig qualityConfig = qualityConfigRepository.all().fetchOne();
```

Comment:
Please create QualityConfigService to fetch the config, throwing an exception if the config is not found,
note that this same config service will be created in PR #11714 so you might have conflict later